lib/rubocop/cop/lint/useless_access_modifier.rb

```diff
@@ -152,7 +160,18 @@ def start_of_new_scope?(child)
 
         def instance_eval_or_new_call?(child)
           class_or_instance_eval?(child) ||
-            class_or_module_or_struct_new_call?(child)
+            class_or_module_or_struct_new_call?(child) ||
+            rails_enabled_and_active_support_new_scope?(child)
+        end
+
+        def rails_enabled_and_active_support_new_scope?(child)
+          if rails_enabled?
+            concerning_block?(child) || class_methods_block?(child)
+          end
```

Comment:
What do you think about simply: `rails_enabled? && (concerning_block?(child) || class_methods_block?(child))`? It has the (albeit, very minor) benefit of returning `false` instead of `nil` if `rails_enabled?` is false